### ISSUE TITLE
[java] UnusedPrivateMethod for methods annotated with jakarta.annotation.PostConstruct or PreDestroy

### ISSUE THREAD

**Issue by kdebski85** [pmd/pmd#5117]
**Affects PMD Version:** 
7.3.0

**Rule:**

[UnusedPrivateMethod](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#unusedprivatemethod)
**Description:**

`UnusedPrivateMethod` should not be reported for methods annotated with  `jakarta.annotation.PostConstruct` or `jakarta.annotation.PreDestroy` since they are usually called by a container (for example Spring).

As a workaround, the annotations can be added to configuration:
```xml
    <rule ref="category/java/bestpractices.xml/UnusedPrivateMethod">
        <properties>
            <property name="ignoredAnnotations" value="java.lang.Deprecated,jakarta.annotation.PostConstruct,jakarta.annotation.PreDestroy" />
        </properties>
    </rule>
```

However, I think such methods should be excluded by default.

**Code Sample demonstrating the issue:**

```java
import jakarta.annotation.PostConstruct;
import jakarta.annotation.PreDestroy;
import org.springframework.stereotype.Component;

@Component
public class JakartaTest {

    @PostConstruct
    private void setup() {
    }

    @PreDestroy
    private void cleanup() {
    }
}
```

**Expected outcome:**

PMD reports a violation at lines 0 and 13, but that's wrong. That's a false positive.

**Running PMD through:** *Gradle*

**Comment by jsotuyod** [comment]
@kdebski85 thanks for the report! this looks like a great candidate for a contribution if you are up to it, I'd expect that same approach + the corresponding tests.

**Comment by kdebski85** [comment]
@jsotuyod I prepared https://github.com/pmd/pmd/pull/5118
Could you please review?
